src/pipeline_func/clean.py

```diff
@@ -0,0 +1,266 @@
+    data['value'] = pd.to_numeric(data['value'], errors='coerce')
+
+    make_note(str(instance)+" instance data loaded")
+
+    new_indic_list = list(VAR_CORR[VAR_CORR['instance'] == instance]['name'])
+    new_df = data[data["dataElement"].isin(new_indic_list)]
+    new_df = pd.DataFrame(new_df.groupby(['dataElement', 'orgUnit', 'period'])[
+                          'value'].sum()).reset_index()
+    new_df = new_df[['period', 'orgUnit', 'value', 'dataElement']]
+
+    make_note(str(instance)+" instance data indicator subset selected")
+
+    return new_df
+
+# Adding composite indicators
+
+
+def get_variable_addition_dict(df, instance):
+    '''build a dict with target vars as keys and original vars to add up as values'''
+
+    target_dict = {}
+
+    df1 = df[df['instance'] == instance]
+    df2 = df1[df1.duplicated('identifier', keep=False)]
+    df3 = df2[['identifier', 'name']]
+
+    for x in list(df3['identifier'].unique()):
+        target_dict[x] = df3[df3['identifier'] == x]['name'].tolist()
+
+    return target_dict
+
+
+def compute_indicators(df, indic_name, indicator_list):
+    '''Compute new vars by summing original vars, and drop original vars '''
+
+    df_new = df[df['dataElement'].isin(indicator_list)]
+    df_new = pd.DataFrame(df_new.groupby(['period', 'orgUnit'])[
+                          'value'].sum()).reset_index()
+    df_new['dataElement'] = indic_name
+
+    df = pd.concat([df, df_new])
+    df.reset_index(drop=True, inplace=True)
+    df = df[~df['dataElement'].isin(indicator_list)]
+
+    return df
+
+# Transforming dates
+
+
+def date_convert(date):
+    if date.find("W") == -1:
+        x = pd.to_datetime(date, format='%Y%m')
+    else:
+        y = date.split("W")[0]
+        w = date.split("W")[1]
+        x = pd.to_datetime(f'{y} {w} 1', format='%G %V %u')
+    return x
+
+
+def process_date(df):
+    df['datetime'] = np.vectorize(date_convert)(df['period'].astype(str))
+    df['year'] = df['datetime'].dt.year
+    df['month'] = df['datetime'].dt.month
+    map_day = {1: 'Jan', 2: 'Feb', 3: 'Mar', 4: 'Apr',
+               5: 'May', 6: 'Jun', 7: 'Jul', 8: 'Aug',
+               9: 'Sep', 10: 'Oct', 11: 'Nov', 12: 'Dec'}
+
+    df['month'] = df['month'].map(map_day)
+    df = df.drop(['datetime', 'period'], axis=1)
+    df['year'] = df['year'].apply(str)
+    return df
+
+#########################
+#     Run functions     #
+#########################
+
+
+def clean(new_dhis_path, old_dhis_path, new_dhis_report_path, old_dhis_report_path):
+
+    new_var_add_dict = get_variable_addition_dict(VAR_CORR, 'new')
+    old_var_add_dict = get_variable_addition_dict(VAR_CORR, 'old')
+
+    # Reporting data
+
+    new_dhis_report_df = get_reporting_data(new_dhis_report_path, "new")
+    make_note('new reporting data formatted for use')
+
+    old_dhis_report_df = get_reporting_data(old_dhis_report_path, "old")
+    make_note('old reporting data formatted for use')
+
```

Comment:
Allocate data resources only when you need them!

If you want this to be less RAM efficient, I would even read-write the data in chunks.

You are doing:

read data 1 
read data 2 
read data 3

process data 1
process data 2 
process data 3

You need to be doing:

read data 1
process data 1
return data 1 or write data 1

read data 2 
process data 2
...